modcp.php

```diff
@@ -3665,7 +3631,7 @@
 		{
 			if(!empty($mybb->input[$input]))
 			{
-				$page_url .= "&amp;{$input}=".htmlspecialchars_uni($mybb->input[$input]);
+				$page_url .= "&amp;{$input}=".urlencode($mybb->input[$input]);
```

Comment:
These lines are for 2 different things - this one is for `array('ipaddress', 'search_users', 'search_posts')` and the one you linked for `array('username', 'sortby', 'order')`.